### ISSUE TITLE
Permission denied during partitioning disks

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#619]
```
root@cmp01:~# parted -m /dev/sda unit B print -s
BYT;
/dev/sda:90028302336B:scsi:512:512:gpt:ATA KINGSTON SVP200S;

ovs@cmp01:/root$ parted -m /dev/sda unit B print -s
Error: Error opening /dev/sda: Permission denied
```

**Comment by kinvaris** [comment]
```
2016-06-16 15:00:46 64900 +0200 - cmp01 - 19875/139717816100672 - celery/celery.worker.job - 188 - ERROR - Task ovs
.storagerouter.configure_disk[8f2da1fd-085d-4691-921e-808bf1ef73a3] raised unexpected: RuntimeError("Command 'parte
d -m /dev/sda unit B print -s' returned non-zero exit status 1 Error: Error opening /dev/sda: Permission denied\n",
)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 297, in new_function
    output = function(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/storagerouter.py", line 1933, in configure_disk
    DiskController.sync_with_reality(storagerouter_guid)
  File "/usr/lib/python2.7/dist-packages/celery/local.py", line 167, in <lambda>
    __call__ = lambda x, *a, **kw: x._get_current_object()(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 439, in __protected_call__
    return orig(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 297, in new_function
    output = function(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/disk.py", line 148, in sync_with_reality
    partitions_info = DiskTools.get_partitions_info(match.groups()[0])
  File "/opt/OpenvStorage/ovs/extensions/generic/disk.py", line 101, in get_partitions_info
    raise RuntimeError('{0} {1}'.format(cpe, cpe.output))
RuntimeError: Command 'parted -m /dev/sda unit B print -s' returned non-zero exit status 1 Error: Error opening /dev/sda: Permission denied
=
```

**Comment by khenderick** [comment]
Duplicates internal ticket OVS-4531

**Comment by wimpers** [comment]
Workaround;

Make sure the partition table is correct before assigning the roles:
- parted /dev/sd[x]
  - Partition Table: loop
- fdisk /dev/sd[x]
  - d / n / p / enter / enter / enter / w
- parted /dev/sd[x]
  - Partition Table: msdos

**Comment by matthiasdeblock** [comment]
Blocker on GIG's environment:

```
2016-08-12 14:58:32 26800 +0200 - stor-01.be-g8-3 - 8576/140272868644672 - celery/celery.worker.job - 5535 - ERROR - Task ovs.storagerouter.configure_disk[faa0fce1-4c10-4e3b-af01-1c6110463c72] raised unexpected: RuntimeError("Command 'parted -m /dev/sda unit B print -s' 
returned non-zero exit status 1 Error: Error opening /dev/sda: Permission denied\n",)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 301, in new_function
    output = function(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/storagerouter.py", line 1806, in configure_disk
    DiskController.sync_with_reality(storagerouter_guid)
  File "/usr/lib/python2.7/dist-packages/celery/local.py", line 167, in <lambda>
    __call__ = lambda x, *a, **kw: x._get_current_object()(*a, **kw)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 439, in __protected_call__
    return orig(self, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/task.py", line 420, in __call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/helpers/decorators.py", line 301, in new_function
    output = function(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/disk.py", line 153, in sync_with_reality
    partitions_info = DiskTools.get_partitions_info(match.groups()[0])
  File "/opt/OpenvStorage/ovs/extensions/generic/disk.py", line 101, in get_partitions_info
    raise RuntimeError('{0} {1}'.format(cpe, cpe.output))
RuntimeError: Command 'parted -m /dev/sda unit B print -s' returned non-zero exit status 1 Error: Error opening /dev/sda: Permission denied
```

**Comment by matthiasdeblock** [comment]
Workaround isn't working as tried it on a gig environment. Needs to be executed as root.

**Comment by kinvaris** [comment]
Even when applying the workaround that @wimpers suggested, it did not work. The problem here is not partition tables but read rights via `parted` command. The command in-code should be executed by `root` when performing `sync-with-reality` not user `ovs` as stated in the initial ticket description.

**Comment by khenderick** [comment]
This was apperently fixed by #833 but the ticket's status wasn't updated

**Comment by JeffreyDevloo** [comment]
### Information

Similar to #674 and #913 

### Test result

Test passed.
Test logic in #913
